libsoftwarecontainer/src/gateway/devicenode/devicenodegateway.h

```diff
@@ -70,6 +70,13 @@ class DeviceNodeGateway :
      */
     virtual bool teardownGateway() override;
 
+    /**
+     * @brief Questions whether the device is configured or not
+     *
+     * return true or false regarding configuration status
+     */
+    virtual bool isDeviceConfigured(const std::string deviceName);
```

Comment:
This function is needed by gtest to examine the idea of "only update device when it is needed".